addons/web_editor/static/src/js/wysiwyg/widgets/link_popover_widget.js

```diff
@@ -254,7 +265,11 @@ const LinkPopoverWidget = Widget.extend({
      */
     _onRemoveLinkClick(ev) {
         ev.preventDefault();
-        this.options.wysiwyg.odooEditor.execCommand('unlink');
+        if (isImg(this.options.wysiwyg.lastElement)) {
+            bus.trigger('deactivate_image_link_tool');
+        } else {
+            this.options.wysiwyg.odooEditor.execCommand('unlink');
+        }
```

Comment:
Same worries as https://github.com/odoo/odoo/pull/84734#discussion_r943613457 for this call too of course... :grimacing:

Reply:
I did not do it initially because I thought it would mean doing something like this: https://github.com/odoo/odoo/commit/0878dcdab1db79b047b29f6feed2bfd270cfc09d.
Instead, I now moved that `if` into a new `removeLink` method of `wysiwyg.js`, and added the missing condition about the snippet menu.